samples/matter/common/src/util/finite_map.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <map>
+
+namespace Nrf
+{
+/*
+   FiniteMap template container allows to wrap mappings between uint16_t-type key and T-type value.
+   The size or the container is predefined at compilation time, therefore
+   the maximum number of stored elements must be well known. FiniteMap owns
+   inserted values, meaning that once the user inserts a value it will not longer
+   be valid outside of the container, i.e. FiniteMap cannot return stored object by copy.
+   FiniteMap's API offers basic operations like:
+     * inserting new values under provided key (Insert)
+     * erasing existing item under given key (Erase) - this operation takes care about the duplicated items,
```

Comment:
Hmm I know this is a left over from my recent modification of this map's code, but we don't automatically care about duplicated items under the hood, as now the responsibility to check that was delegated to the application (with `GetDuplicatesCount()`). Currently `Erase` is suppose to erase an element unconditionally.